fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ReviewQualityCheckScaffold.kt

```diff
@@ -86,18 +89,23 @@ fun ReviewQualityCheckScaffold(
 }
 
 @Composable
-private fun Header() {
+private fun Header(modifier: Modifier) {
+    val reviewCheckerText = stringResource(R.string.review_quality_check_feature_name_2)
+    val betaText = stringResource(R.string.review_quality_check_beta_flag)
+    val heading = stringResource(id = R.string.a11y_heading)
+
     Row(
-        modifier = Modifier.semantics(mergeDescendants = true) {},
+        modifier = modifier
+            .clearAndSetSemantics { contentDescription = "$reviewCheckerText, $betaText, $heading" },
         verticalAlignment = Alignment.CenterVertically,
     ) {
         Text(
-            text = stringResource(R.string.review_quality_check_feature_name_2),
+            text = reviewCheckerText,
             color = FirefoxTheme.colors.textPrimary,
             style = FirefoxTheme.typography.headline6,
         )
 
-        Spacer(modifier = Modifier.width(8.dp))
+        Spacer(modifier = modifier.width(8.dp))
```

Comment:
Try not to reuse passed in modifiers in different composables, this should be resolved as Header doesn't need a modifier param.